torch/testing/_internal/opinfo/definitions/signal.py

```diff
@@ -0,0 +1,269 @@
+import random
+import unittest
+
+from itertools import product
+from typing import List
+
+import torch
+from torch.testing._internal.common_dtype import all_types_and
+from torch.testing._internal.common_utils import TEST_SCIPY
+from torch.testing._internal.opinfo.core import (
+    DecorateInfo,
+    ErrorInput,
+    OpInfo,
+    SampleInput,
+)
+
+if TEST_SCIPY:
+    import scipy.signal
+
+
+def sample_inputs_window(op_info, device, dtype, requires_grad, *, kws=({},), **kwargs):
+    _kwargs = dict(
+        kwargs, **{"device": device, "dtype": dtype, "requires_grad": requires_grad}
+    )
+    sizes = [0, 1, 2, 5, 10, 50, 100, 1024, 2048]
```

Comment:
Sample inputs should be smaller tensors that verify interesting arguments to the function. They're going to be used to test the correctness of almost two dozen systems!

As @lezcano says, an OpInfo can also define reference inputs (which are typically all the sample inputs + additional larger inputs). Tests can then optionally use the expanded set of reference inputs, and the NumPy/SciPy consistency test uses them:

https://github.com/pytorch/pytorch/blob/3f2e7d5c9a5569e4c2d4857d01697fc2bfbfe4fa/test/test_ops.py#L161

Reply:
Gotcha! I'll be working on it tonight.